app/ts/simulation/services/EthereumClientService.ts

```diff
@@ -250,9 +259,9 @@ export class EthereumClientService {
 		}
 		const erc20ABI = ['event Transfer(address indexed from, address indexed to, uint256 value)']
 		const erc20 = new ethers.Interface(erc20ABI)
-		const flattenedLogs = events.flat() 
+		const flattenedLogs = events.flat()
 		const parsedEthLogs = parseEthLogs(flattenedLogs)
-		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(log => log.args[0]).concat(parsedEthLogs.map(log => log.args[1]).concat(senders)))
+		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(log => EthereumAddress.parse(log.args[0])).concat(parsedEthLogs.map(log => EthereumAddress.parse(log.args[1])).concat(senders)))
```

Comment:
Consider extracting these lambdas out into named functions on a previous line for improved clarity:
```suggestion
		const extractLogTopic = (log: TodoTypeHere) => EthereumAddress.parse(log.args[0])
		const extractLogSender = (log: TodoTypeHere) => EthereumAddress.parse(log.args[1])
		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(extractLogTopic).concat(parsedEthLogs.map(extractLogSender).concat(senders)))
```
I'm not sure I got these names right, they are meant as an example.